snapshotter/main.cc

```diff
@@ -121,6 +122,49 @@ Dart_Handle HandleLibraryTag(Dart_LibraryTag tag,
   return GetLoader().HandleLibraryTag(tag, library, url);
 }
 
+static const char StubNativeFunctionName[] = "StubNativeFunction";
+
+void StubNativeFunction(Dart_NativeArguments arguments) {
+  // This is a stub function for the resolver
+  Dart_SetReturnValue(
+      arguments, Dart_NewApiError("<EMBEDDER DID NOT SETUP NATIVE RESOLVER>"));
+}
+
+static Dart_NativeFunction StubNativeLookup(Dart_Handle name,
+                                            int argument_count,
+                                            bool* auto_setup_scope) {
+  return &StubNativeFunction;
+}
+
+static const uint8_t* StubNativeSymbol(Dart_NativeFunction nf) {
+  return reinterpret_cast<const uint8_t*>(StubNativeFunctionName);
+}
+
+// Registers a dummy native symbol resolver on all loaded libraries that do not
+// already have a native symbol resolver. This is necessary because
+// `--compile-all` will try and resolve all native functions. The function
+// returned by the dummy native symbol resolver will never be invoked.
+static void SetupStubNativeResolvers() {
+  Dart_Handle library_ids = Dart_GetLibraryIds();
+  intptr_t library_ids_length;
+  Dart_ListLength(library_ids, &library_ids_length);
```

Comment:
I guess error checking is not performed for any of these calls. I'd rather escape out of this routine on any error. Maybe a macro or something. But if you think that is being too paranoid, I understand. I don't know how these `dart_api.h` calls can fail.